main/AppTask.cpp

```diff
@@ -481,6 +506,170 @@ void AppTask::DispatchEvent(AppEvent * aEvent)
+                                                                                    aInParam.QueryPrepareFailed.StatusReport->mStatusCode) :
+                                                                 "None");
+            break;
+        }
+
+        case SoftwareUpdateManager::kEvent_SoftwareUpdateAvailable:
+        {
+            NRF_LOG_INFO("Software Update Available - Priority: %d Condition: %d Version: %s IntegrityType: %d URI: %s",
+                                                                                aInParam.SoftwareUpdateAvailable.Priority,
+                                                                                aInParam.SoftwareUpdateAvailable.Condition,
+                                                                                aInParam.SoftwareUpdateAvailable.Version,
+                                                                                aInParam.SoftwareUpdateAvailable.IntegrityType,
+                                                                                aInParam.SoftwareUpdateAvailable.URI);
+
+            break;
+        }
+
+        case SoftwareUpdateManager::kEvent_FetchPartialImageInfo:
+        {
+            aOutParam.FetchPartialImageInfo.PartialImageLen = 0;
+            break;
+        }
+
+        case SoftwareUpdateManager::kEvent_ClearImageFromStorage:
+        {
+            NRF_LOG_INFO("Clearing Image Storage");
+            numBytesDownloaded = 0;
+            break;
+        }
+
+        case SoftwareUpdateManager::kEvent_StartImageDownload:
+        {
+            sSHA256.Begin();
+            NRF_LOG_INFO("Starting Image Download");
+            break;
+        }
+        case SoftwareUpdateManager::kEvent_StoreImageBlock:
+        {
+            sSHA256.AddData(aInParam.StoreImageBlock.DataBlock, aInParam.StoreImageBlock.DataBlockLen);
+            numBytesDownloaded += aInParam.StoreImageBlock.DataBlockLen;
+            break;
+        }
+
+        case SoftwareUpdateManager::kEvent_ComputeImageIntegrity:
+        {
+            NRF_LOG_INFO("Num. bytes downloaded: %d", numBytesDownloaded);
+
+            // Make sure that the buffer provided in the parameter is large enough.
+            if (aInParam.ComputeImageIntegrity.IntegrityValueBufLen < sSHA256.kHashLength)
+            {
+                aOutParam.ComputeImageIntegrity.Error = WEAVE_ERROR_BUFFER_TOO_SMALL;
+            }
+            else
+            {
+                sSHA256.Finish(aInParam.ComputeImageIntegrity.IntegrityValueBuf);
+            }
+            break;
+        }
+
+        case SoftwareUpdateManager::kEvent_ReadyToInstall:
+        {
+            NRF_LOG_INFO("Image is ready to be installed");
+            break;
+        }
+
+        case SoftwareUpdateManager::kEvent_StartInstallImage:
+        {
+            NRF_LOG_INFO("Image Install is not supported in this example application");
+            SoftwareUpdateMgr().Abort();
```

Comment:
Seems like the example should declare the image as having been installed, rather than abort.

Reply:
The Generic Impl returns WEAVE_ERROR_NOT_IMPLEMENTED when InstallImage method is called. This would cause software update to fail with error WEAVE_ERROR_NOT_IMPLEMENTED and invoke the retry logic. To aboid that happening for the lock example, I added Abort here so that the attempt finishes more gracefully.
I could override InstallImage method in concrete impl for nrf patform and make it return WEAVE_NO_ERROR so that the installation step goes through without errors. Would you be in favor of that ?

Reply:
As we discussed offline, I made sure that the application can call ImageInstallationComplete to mark the completion of the install stage and to move our state back to idle.